src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailsChart.tsx

```diff
@@ -0,0 +1,144 @@
+import { ReactComponent as InfoIcon } from '../../../../../assets/icons/info-filled.svg'
+import { ReactComponent as Chat } from '../../../../../assets/icons/ic-chat-circle-dots.svg'
+import { APP_STATUS_HEADERS, DEPLOYMENT_STATUS } from '../../../../../config'
+import { StatusFilterButtonComponent } from '../../k8Resource/StatusFilterButton.component'
+import IndexStore from '../../index.store'
+import { NodeStreamMap } from '../environment.type'
+import { AggregatedNodes } from '../../../../app/types'
+import { aggregateNodes } from '../../../../app/details/appDetails/utils'
+import { STATUS_SORTING_ORDER } from './constants'
+
+export default function AppStatusDetailsChart({ appStreamData, filterRemoveHealth = false, showFooter }) {
+    const _appDetails = IndexStore.getAppDetails()
+    const [nodeStatusMap, setNodeStatusMap] = useState<Map<string, NodeStreamMap>>()
+    const [currentFilter, setCurrentFilter] = useState('')
+
+    useEffect(() => {
+        try {
+            const stats = appStreamData.result.application.status.operationState.syncResult.resources.reduce(
+                (agg, curr) => {
+                    agg.set(`${curr.kind}/${curr.name}`, curr)
+                    return agg
+                },
+                new Map(),
+            )
+            setNodeStatusMap(stats)
+        } catch (error) {}
+    }, [appStreamData])
+
+    const nodes: AggregatedNodes = useMemo(() => {
+        return aggregateNodes(_appDetails.resourceTree?.nodes || [], _appDetails.resourceTree?.podMetadata || [])
+    }, [_appDetails])
+    const nodesKeyArray = Object.keys(nodes?.nodes || {})
+    let flattenedNodes = []
+    if (nodesKeyArray.length > 0) {
+        for (let index = 0; index < nodesKeyArray.length; index++) {
+            const element = nodes.nodes[nodesKeyArray[index]]
+            element.forEach((childElement) => {
+                childElement.health && flattenedNodes.push(childElement)
+            })
+        }
+        flattenedNodes.sort((a, b) => {
+            return (
+                STATUS_SORTING_ORDER[a.health.status?.toLowerCase()] -
+                STATUS_SORTING_ORDER[b.health.status?.toLowerCase()]
+            )
+        })
+    }
+
+    function getNodeMessage(kind: string, name: string) {
+        if (nodeStatusMap && nodeStatusMap.has(`${kind}/${name}`)) {
+            const { message } = nodeStatusMap.get(`${kind}/${name}`)
+            return message
+        }
+        return ''
+    }
+
+    const onFilterClick = (selectedFilter: string): void => {
+        if (currentFilter !== selectedFilter.toLowerCase()) {
+            setCurrentFilter(selectedFilter.toLowerCase())
+        }
+    }
+
+    const filteredFlattendNodes = () => {
+        if (filterRemoveHealth) {
+            return flattenedNodes?.filter((node) => node.health.status?.toLowerCase() !== DEPLOYMENT_STATUS.HEALTHY) || []
+        } else {
+            return flattenedNodes || []
+        }
```

Comment:
move this out of this function & create a local var

Reply:
done